bothan-coingecko/src/worker/asset_worker.rs

```diff
@@ -0,0 +1,150 @@
+use std::sync::Weak;
+use std::time::Duration;
+
+use chrono::NaiveDateTime;
+use rust_decimal::prelude::FromPrimitive;
+use rust_decimal::Decimal;
+use tokio::time::{interval, timeout, Interval};
+use tracing::{debug, info, warn};
+
+use bothan_core::store::Store;
+use bothan_core::types::AssetInfo;
+
+use crate::api::rest::CoinGeckoRestAPI;
+use crate::api::types::{Market, Order};
+use crate::worker::error::ParseError;
+use crate::worker::CoinGeckoWorker;
+
+pub(crate) fn start_asset_worker(
+    weak_worker: Weak<CoinGeckoWorker>,
+    mut update_interval: Interval,
```

Comment:
Umm, can we receive `Duration` from caller and create `Interval` in this function instead?